x-pack/platform/plugins/shared/lens/public/datasources/form_based/to_esql.ts

```diff
@@ -145,10 +145,13 @@ export function getESQLForLayer(
 
     if (wrapInFilter || wrapInTimeFilter) {
       if (wrapInFilter) {
-        if (col.filter?.language === 'kquery') {
+        if (col.filter?.language === 'kuery') {
+          metricESQL += ` WHERE KQL("""${col.filter.query}""")`;
```

Comment:
Is it just safe like that with the triple quote or we need to escape the query filter?

Reply:
i added removal of all """ inside the filter.query